### PR TITLE
Add a ColorWidget

### LINKED ISSUE #165: Add a ColorWidget.
Spec https://html.spec.whatwg.org/multipage/forms.html#color-state-%28type=color%29

Creating an alternative widget for older browsers is probably not needed as a text input is probably good enough.

### PR DESCRIPTION
**Pull request by magopian**
Fixes #165 

On Firefox:
screen shot 2016-04-25 at 16 26 01

On Chrome:
screen shot 2016-04-25 at 16 26 39

### CODE REVIEW

File: src/components/widgets/ColorWidget.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+if (process.env.NODE_ENV !== "production") {
+  ColorWidget.propTypes = {
+    schema: PropTypes.object.isRequired,
+    id: PropTypes.string.isRequired,
+    placeholder: PropTypes.string,
+    value: PropTypes.oneOfType([
+      React.PropTypes.string,
+      React.PropTypes.number,
```
Comment (n1k0) [inline src/components/widgets/ColorWidget.js]: Could this ever be a number?

File: src/components/widgets/ColorWidget.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      required={required}
+      onChange={(event) => onChange(event.target.value)} />
+  );
+}
+
+if (process.env.NODE_ENV !== "production") {
+  ColorWidget.propTypes = {
+    schema: PropTypes.object.isRequired,
+    id: PropTypes.string.isRequired,
+    placeholder: PropTypes.string,
```
Comment (n1k0) [inline src/components/widgets/ColorWidget.js]: I don't think this input supports placeholders, we should drop support for this prop for this widget.

File: test/uiSchema_test.js [hunk 3]
```diff
@@ -195,6 +195,41 @@ describe("uiSchema", () => {
...
+      it("should accept a uiSchema object", () => {
+        const {node} = createFormComponent({schema, uiSchema});
+
+        expect(node.querySelectorAll("[type=color]"))
+          .to.have.length.of(1);
+      });
+
+      it("should support formData", () => {
+        const {node} = createFormComponent({schema, uiSchema, formData: {
+          foo: "a"
```
Comment (n1k0) [inline test/uiSchema_test.js]: Maybe we should add a really color value here?

File: src/components/widgets/ColorWidget.js (hunk 2 above)
Comment (magopian) [inline src/components/widgets/ColorWidget.js]: too much copy/paste right? :D I plead guilty

File: test/uiSchema_test.js [hunk 4]
```diff
@@ -195,6 +195,41 @@ describe("uiSchema", () => {
...
+        const {comp, node} = createFormComponent({schema, uiSchema, formData: {
+          foo: "a"
+        }});
+
+        return SimulateAsync().change(node.querySelector("[type=color]"), {
+          target: {value: "b"}
+        })
+          .then(() => expect(comp.state.formData).eql({foo: "b"}));
+      });
+    });
```
Comment (n1k0) [inline test/uiSchema_test.js]: I'd love seeing a test for failing validation as well.

**Comment by n1k0** [comment]
LGTM, r+wc

File: test/uiSchema_test.js (hunk 4 above)
Comment (magopian) [inline test/uiSchema_test.js]: What do you mean by "failing validation"? It's only a widget, should it do any validation?

Comment (n1k0) [inline test/uiSchema_test.js]: Oh, it's an uiSchema test, I see. Well so we need to add some test in StringField_test to check for the field behavior, like validation (look at what we did for date widgets for instance).

**Comment by n1k0** [comment]
r+

File: src/components/widgets/ColorWidget.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  );
+}
+
+if (process.env.NODE_ENV !== "production") {
+  ColorWidget.propTypes = {
+    schema: PropTypes.object.isRequired,
+    id: PropTypes.string.isRequired,
+    value: PropTypes.oneOfType([
+      React.PropTypes.string
+    ]),
```
Comment (n1k0) [inline src/components/widgets/ColorWidget.js]: `oneOfType` is not needed anymore and could be removed.

**Pull request merged**